glass/src/lib/native/cpp/hardware/SpeedController.cpp

```diff
@@ -0,0 +1,32 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "glass/hardware/SpeedController.h"
+
+#include <imgui.h>
+
+#include "glass/Context.h"
+#include "glass/DataSource.h"
+
+using namespace glass;
+
+void glass::DisplaySpeedController(SpeedControllerModel* m) {
+  // Get duty cycle data from the model and do not display anything if the data
+  // is null.
```

Comment:
It feels unnatural to hide the widget completely if no data is available, because then they "pop" into existence when the data becomes available.  What we do in other places is grey out the control / display "unknown" if the data isn't available.

Reply:
This is part of the reason for the model Exists() function too, see e.g. DIO.cpp.